flink-ml-lib/src/main/java/org/apache/flink/ml/feature/univariatefeatureselector/UnivariateFeatureSelectorModel.java

```diff
@@ -0,0 +1,201 @@
+
+        final String broadcastModelKey = "broadcastModelKey";
+        RowTypeInfo inputTypeInfo = TableUtils.getRowTypeInfo(inputs[0].getResolvedSchema());
+        RowTypeInfo outputTypeInfo =
+                new RowTypeInfo(
+                        ArrayUtils.addAll(inputTypeInfo.getFieldTypes(), VectorTypeInfo.INSTANCE),
+                        ArrayUtils.addAll(inputTypeInfo.getFieldNames(), getOutputCol()));
+
+        DataStream<Row> outputStream =
+                BroadcastUtils.withBroadcastStream(
+                        Collections.singletonList(data),
+                        Collections.singletonMap(broadcastModelKey, modelData),
+                        inputList -> {
+                            DataStream input = inputList.get(0);
+                            return input.map(
+                                    new PredictOutputFunction(getFeaturesCol(), broadcastModelKey),
+                                    outputTypeInfo);
+                        });
+
+        return new Table[] {tEnv.fromDataStream(outputStream)};
+    }
+
+    /** This operator loads model data and predicts result. */
+    private static class PredictOutputFunction extends RichMapFunction<Row, Row> {
+
+        private final String inputCol;
+        private final String broadcastKey;
+        private int[] indices;
+
+        public PredictOutputFunction(String inputCol, String broadcastKey) {
+            this.inputCol = inputCol;
+            this.broadcastKey = broadcastKey;
+        }
+
+        @Override
+        public Row map(Row row) {
+            if (indices == null) {
+                UnivariateFeatureSelectorModelData modelData =
+                        (UnivariateFeatureSelectorModelData)
+                                getRuntimeContext().getBroadcastVariable(broadcastKey).get(0);
+                indices = Arrays.stream(modelData.indices).sorted().toArray();
+            }
+
+            if (indices.length == 0) {
+                return Row.join(row, Row.of(Vectors.dense()));
+            } else {
+                Vector inputVec = ((Vector) row.getField(inputCol));
+                Preconditions.checkArgument(
+                        inputVec.size() > indices[indices.length - 1],
+                        "Input %s features, but UnivariateFeatureSelector is "
+                                + "expecting at least %s features as input.",
+                        inputVec.size(),
+                        indices[indices.length - 1] + 1);
+                Vector outputVec = selectByIndices(inputVec, indices);
+                return Row.join(row, Row.of(outputVec));
+            }
+        }
+
+        /**
+         * Selects a subset of the vector base on the indices. Note that the input indices must be
+         * sorted in ascending order if the input vector is sparse.
+         */
+        private Vector selectByIndices(Vector vector, int[] selectedIndices) {
```

Comment:
Since this method looks pretty generic and it is re-used by `VarianceThresholdSelectorModel.java`, how about we move this method to `org.apache.flink.ml.common.util.VectorUtils` and make it static?

Reply:
I prefer to keep this method private because this method requires the input indices to be sorted but without any check for better performance, so I don't want to expose it as public for only two operators.

Reply:
`DataStreamUtils#sample(...)` are also called only by two classes.

It is OK for methods to assume that the input meets certain criteria (e.g. not null). Requiring the inputs to be sorted is just one special requirement and we can document it properly in the util method's Java doc. We can also name the variable as `sortedIndices` to make the requirement self-explanatory.

The benefit of adding util method is to reduce duplicated code. Is there other downside of making it util method?